common/achievement/Readme.md

```diff
@@ -1,4 +1,352 @@
+| `evaluate.ts`                         | functions to evaluate if an achievements conditions are met based on the [conditionDataAggregation](#conditionDataAggregation) of the related achievement_template                                                                   |
+| `get.ts`                              | resolver functions to get all achievements, get certain achievements to be displayed as next steps, get achievements that are yet to be initialized, get achievements by their id and assemble the achievement data for the frontend |
+| `index.ts`                            | functions to initialize the evaluation process when an action was taken, cache metrics for events, and reward users when achievement conditions are met                                                                              |
+| <div id="metric.ts">`metric.ts`</div> | functions to create metrics together with their related actions when building the project                                                                                                                                            |
+| `metrics.ts`                          | functions to register one or more metrics and to get metrics based on an action id                                                                                                                                                   |
+| `template.ts`                         | functions to get all templates presorted by group or metrics and to get templates with a certain metric                                                                                                                              |
+| `types.ts`                            | additional types used for the process of evaluation and creation of achievements                                                                                                                                                     |
+| `util.ts`                             | additional functions for achievement processing                                                                                                                                                                                      |
+
+### 1.3. Resolvers
+
+All resolvers can be found in `common/achivement/get.ts`
+
+1.  `getAchievementById`: function to get an achievement of a user based on an achievement id
+2.  `getNextStepAchievements`: function to get a users achievements that will be displayed as a next step card on their dashboard
+3.  `getFurtherAchievements`: function to get the achievement templates, which do not yet have an associated user achievement for the corresponding user but belong to an already initialized achievement group, as inactive achievements.
+4.  `getUserAchievements`: function to get all user achievements found for a user
+
+## 2. Achievement System
+
+The achievement system is set up as follows
+
+```mermaid
+flowchart TD
```

Comment:
I like the idea of flowcharts ❤️ 

I would suggest focusing more on the concepts rather than simply mirroring what the code does. For instance, in the code, we have a wrapper function (isAchievementConditionMet) around evaluateAchievement. While it may make sense in that context, it doesn't add value to this diagram. The diagram should aid in understanding the underlying idea; the specifics can be found in the code, and these are likely to change over time.

Reply:
I removed the explainations of the functions and just kept the basic process.